### PR TITLE
Move auto login check to useEffect

### PR DESCRIPTION
**Pull request by trishalim**

### CODE REVIEW

File: packages/jazz-react/src/auth/DemoAuth.tsx [hunk 1]
```diff
@@ -81,16 +81,19 @@ export const DemoAuthBasicUI = ({
       ? window.matchMedia("(prefers-color-scheme: dark)").matches
       : false;
 
-  if (user && state.state === "ready") {
+  const isAutoLogin = !!(user && state.state === "ready");
```
Comment (trishalim) [inline packages/jazz-react/src/auth/DemoAuth.tsx:84]: @gdorsi The `Boolean()` gave me TS errors that don't make sense to me
image

Comment (gdorsi) [reply packages/jazz-react/src/auth/DemoAuth.tsx:84]: Ah, it's because Typescript doesn't do type narrowing when using `Boolean`.

**Review by gdorsi — APPROVED** [review]
Looks good to me!

**Comment by aeplay** [comment]
Thanks for the advice @gdorsi and for the updated implementation @trishalim !

**Pull request merged**